src/cargo/core/compiler/mod.rs

```diff
@@ -640,13 +640,17 @@ fn rustdoc(cx: &mut Context<'_, '_>, unit: &Unit) -> CargoResult<Work> {
     rustdoc.arg("-C").arg(format!("metadata={}", metadata));
 
     let scrape_output_path = |unit: &Unit| -> CargoResult<PathBuf> {
-        let output_dir = cx.files().deps_dir(unit);
-        Ok(output_dir.join(format!("{}.examples", unit.buildkey())))
+        cx.outputs(unit)
+            .map(|outputs| outputs[0].path.to_path_buf())
     };
 
     if unit.mode.is_doc_scrape() {
         debug_assert!(cx.bcx.scrape_units.contains(unit));
 
+        if unit.target.is_test() {
+            rustdoc.arg("--scrape-tests");
+        }
```

Comment:
It isn't possible in that you can't pass `--tests` to `cargo doc`. But re: later in your review, if we make it so `[[tests]] doc-scrape-examples = true` will proactively include the test target, then this could arise. So it's a bit of future proofing.